src/dataflow/components/dataflow-program-topbar.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from "react";
+import { ProgramRunTime, ProgramRunTimes } from "../utilities/node";
+
+import "./dataflow-program-topbar.sass";
+
+interface IProps {
+  onRunProgramClick: () => void;
+  onStopProgramClick: () => void;
+  onProgramTimeSelectClick: (type: number) => void;
+  isRunEnabled: boolean;
+}
+
+export class DataflowProgramTopbar extends React.Component<IProps, {}> {
+  constructor(props: IProps) {
+    super(props);
+  }
+
+  public render() {
+    return (
+      <div className="program-editor-topbar">
+        <div>Duration:</div>
+        <select
+          onChange={this.handleSelectChange}
+          disabled={this.props.isRunEnabled}
+        >
+          { ProgramRunTimes.map((rt: ProgramRunTime, i: number) => (
+              <option key={i} value={rt.val}>
```

Comment:
Using index for key is not recommended. In this case I'd use `rt.text`.

Reply:
Fixed.